WizardWorldApi/WizardWorldApi.Tests.Integrations/Tests/SpellControllerTests.cs

```diff
@@ -26,25 +28,125 @@ public class SpellControllerTests {
         }
 
         [Test]
-        public async Task Get_ShouldReturnSpellsList() {
+        public async Task Get_NoQueryParameters_ShouldReturnSpellsList() {
             // Arrange
             var expectedSpells = SpellsGenerator.Spells;
             // Act 
             var response = await _client.GetAsync("Spell");
             response.StatusCode.Should().Be(HttpStatusCode.OK);
-            var spells = await response.Content.ReadFromJsonAsync<List<SpellDto>>();
+            var spells = await response.Content.DeserializeAsync<List<SpellDto>>();
             // Assert
             spells.Should().NotBeEmpty();
             spells.Should().BeEquivalentTo(expectedSpells);
         }
+        
+        [Test]
+        public async Task Get_Name_ShouldReturnSpellsList() {
+            // Arrange
+            var expectedSpells = SpellsGenerator.Spells;
+            var expectedSpell = expectedSpells.First();
+            // Act 
+            var response = await _client.GetAsync($"Spell?Name={expectedSpell.Name}");
+            response.StatusCode.Should().Be(HttpStatusCode.OK);
+            var spells = await response.Content.DeserializeAsync<List<SpellDto>>();
+            // Assert
+            spells.Should().NotBeEmpty();
+            spells[0].Should().BeEquivalentTo(expectedSpell);
```

Comment:
Filter `expectedSpells` with `.Where` and compare two lists using `Should().BeEquavalentOf(exectedSpells)`